File: ts/two_sum.ts
function twoSum_(nums: number[], target: number): number[] {
  const hash = new Map();

  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i];

    if (hash.has(complement)) {
      return [hash.get(complement), i];
    }

    hash.set(nums[i], i);
  }
  return [];
}

File: easy/python/two_sum.py
"""Two Sum problem solution.

This module contains a simple hash-map based solution to the classic
Two Sum problem. The original implementation imported ``List`` from the
``ast`` module which is incorrect; the type resides in ``typing``.
"""

from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup = {}
        for idx, num in enumerate(nums):
            if num in lookup:
                return [lookup[num], idx]
            lookup[target - num] = idx

        return []
